### PR TITLE
Valid pem format to the CRDs webhooks instead empty placeholder

### PR DESCRIPTION
**Pull request by cyclinder**
We need to have a valid format and not only the placeholder, as the apiserver will refuse deletion with "unable to load root certificates: unable to parse bytes as PEM block"

Fixed https://github.com/metallb/metallb/issues/1501
Fixed https://github.com/metallb/metallb/issues/1521

### CODE REVIEW

**Comment by fedepaol** [comment]
We are trying to have the commit messages as per https://cbea.ms/git-commit/ 

Can you change the commit with something like:

Add a valid pem format to the CRDs webhooks instead of the empty placeholder

We need to have a valid format and not only the placeholder, as the apiserver will refuse deletion with "unable to load root certificates: unable to parse bytes as PEM block" ?

**Comment by fedepaol** [comment]
Also, can you add another commit to track this change to the release notes here https://github.com/metallb/metallb/blob/a942f290629bf883f76d7636066b895f8111929f/website/content/release-notes/_index.md#L6 ?

**Comment by cyclinder** [comment]
Thanks @fedepaol

**Comment by cyclinder** [comment]
I found "`First-time contributors need a maintainer to approve running workflows`,"  Could you enable CI for the PR ?

File: website/content/release-notes/_index.md [hunk 1]
```diff
@@ -13,7 +13,10 @@ Bug Fixes:
 the service ([Issue 1511](https://github.com/metallb/metallb/issues/1511) [PR 1516](https://github.com/metallb/metallb/pull/1516))
 - Add the AvoidBuggyIPs flag to the IPAddressPool CRD. Converting a CIDR to a range comes with limitation related
 to setting the aggregation length and validating it. ([Issue 1495](https://github.com/metallb/metallb/issues/1495),
-[PR 1515](https://github.com/metallb/metallb/pull/1515)
+[PR 1515](https://github.com/metallb/metallb/pull/1515))
+- Add a valid pem format to the CRDs webhooks instead of the empty placeholder. ([Issue 1501](https://github.com/metallb/metallb/issues/1501),
```
Comment (fedepaol) [inline website/content/release-notes/_index.md:17]: Can you add this change in a separate commit?

Comment (cyclinder) [reply website/content/release-notes/_index.md:17]: Do you mean to make this change in another PR?

**Comment by fedepaol** [comment]
> We are trying to have the commit messages as per https://cbea.ms/git-commit/
> 
> Can you change the commit with something like:
> 
> Add a valid pem format to the CRDs webhooks instead of the empty placeholder
> 
> We need to have a valid format and not only the placeholder, as the apiserver will refuse deletion with "unable to load root certificates: unable to parse bytes as PEM block" ?

Sorry I wasn't clear. What I meant above was also to change the commit body with the second part here ^^^^ : 
```
We need to have a valid format and not only the placeholder, as the apiserver will refuse deletion with "unable to load root certificates: unable to parse bytes as PEM block
```

Also, the commit title is too long and gets cut, please try to shorten it (Valid pem format to the CRDs webhooks instead empty placeholder should work)

File: website/content/release-notes/_index.md (hunk 1 above)
Comment (fedepaol) [reply website/content/release-notes/_index.md:17]: nope, I mean a different commit in the same pr (you can have multiple commits in your https://github.com/cyclinder/metallb/tree/fix_invalid_caBundle branch, those will be reflected in separate commits in the pr)

**Comment by cyclinder** [comment]
> > We are trying to have the commit messages as per https://cbea.ms/git-commit/
> > Can you change the commit with something like:
> > Add a valid pem format to the CRDs webhooks instead of the empty placeholder
> > We need to have a valid format and not only the placeholder, as the apiserver will refuse deletion with "unable to load root certificates: unable to parse bytes as PEM block" ?
> 
> Sorry I wasn't clear. What I meant above was also to change the commit body with the second part here ^^^^ :
> 
> ```
> We need to have a valid format and not only the placeholder, as the apiserver will refuse deletion with "unable to load root certificates: unable to parse bytes as PEM block
> ```
> 
> Also, the commit title is too long and gets cut, please try to shorten it (Valid pem format to the CRDs webhooks instead empty placeholder should work)

If my understand is right：
```
PR title: "Valid pem format to the CRDs webhooks instead empty placeholder should work",
First commit message: "Valid pem format to the CRDs webhooks instead empty placeholder should work",
Second commit message: "Release notes: Add a valid pem format to the CRDs webhooks instead of the empty placeholder"
```
Is this ok?

**Comment by fedepaol** [comment]
> > > We are trying to have the commit messages as per https://cbea.ms/git-commit/
> > > Can you change the commit with something like:
> > > Add a valid pem format to the CRDs webhooks instead of the empty placeholder
> > > We need to have a valid format and not only the placeholder, as the apiserver will refuse deletion with "unable to load root certificates: unable to parse bytes as PEM block" ?
> > 
> > 
> > Sorry I wasn't clear. What I meant above was also to change the commit body with the second part here ^^^^ :
> > ```
> > We need to have a valid format and not only the placeholder, as the apiserver will refuse deletion with "unable to load root certificates: unable to parse bytes as PEM block
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Also, the commit title is too long and gets cut, please try to shorten it (Valid pem format to the CRDs webhooks instead empty placeholder should work)
> 
> If my understand is right：
> 
> ```
> PR title: "Valid pem format to the CRDs webhooks instead empty placeholder should work",
> First commit message: "Valid pem format to the CRDs webhooks instead empty placeholder should work",
> Second commit message: "Release notes: Add a valid pem format to the CRDs webhooks instead of the empty placeholder"
> ```
> 
> Is this ok?

 PR title: "Valid pem format to the CRDs webhooks instead empty placeholder",
First commit message: "Valid pem format to the CRDs webhooks instead empty placeholder",
 Second commit message: "Release notes: mention the cabundle placeholder fix"

Also: add the suggested body: 
```
We need to have a valid format and not only the placeholder, as the apiserver will refuse deletion with "unable to load root certificates: unable to parse bytes as PEM block.
```
to the first commit

**Comment by cyclinder** [comment]
Thanks! And how to run `inv generatemanifests`?  I don't found this command,  I'm sorry for this is my first PR for `metallb`..

**Comment by cyclinder** [comment]
@fedepaol Thanks, Updated. Let's go CI test.

**Comment by fedepaol** [comment]
> @fedepaol Thanks, Updated. Is the `Run inv generatemanifests` required?
> 
> ```
> root@master:~/metallb# inv generatemanifests
> /bin/bash: line 1: controller-gen: command not found
> ```

nope, I see you changed the manifests manually. CI will complain if they are not consistent with the generated result.

**Comment by fedepaol** [comment]
The requirements for dev are described here https://metallb.universe.tf/community/#required-software

**Comment by fedepaol** [comment]
@cyclinder so, CI failed for some difference between the generated manifests under `https://github.com/metallb/metallb/tree/main/config/manifests` which are generated from the kustomize configuration (including https://github.com/metallb/metallb/blob/main/config/crd/crd-conversion-patch.yaml which you changed).
In order to get it straight, you need to download and install invoke (https://www.pyinvoke.org/) and run `inv generatemanifests` as CI suggests (please amend the first commit and not add a new one)

**Comment by fedepaol** [comment]
Also, come to slack if you need help with that

**Comment by fedepaol** [comment]
LGTM, thanks!

**Pull request merged**